mantidproject/mantid
# Muon undo fit resets correctly

**Description of work.**
In the muon GUI it was possible to cause a crash by undoing the fit to a state that is not compatible with the current function.
This PR fixes that issue by resetting the undo history when the functional form is changed.



**To test:**


Open Muon analysis
Load some data
Go to fitting
Add a function
Click fit
Click undo - should work as expected
Click fit

Add another function
Check that undo is reset to "0"
Do a fit

Remove a function
Check that undo is reset to "0"
Do a fit

Remove the last function
Check that undo is reset to "0"


Fixes #32798. 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
